packages/forms-web-app/src/lib/dashboard-functions.js

```diff
@@ -39,8 +107,98 @@ const formatAppealType = (caseDataAppealType) => {
 	}
 };
 
+/**
+ * @param {LPAAppealData} appealCaseData return object from database call
+ * @returns {boolean} returns depending on whether a Questionnaire due date has been set
+ */
+
+const isNewAppeal = (appealCaseData) => {
+	return !appealCaseData.questionnaireDueDate;
+};
+
+/**
+ * @param {LPAAppealData} appealCaseData return object from database call
+ * @returns {DueDocumentType} object containing details of next due document
+ */
+const determineDocumentToDisplayLPADashboard = (appealCaseData) => {
+	if (isQuestionnaireDue(appealCaseData)) {
+		return {
+			deadline: appealCaseData.questionnaireDueDate,
+			dueInDays: calculateDueInDays(appealCaseData.questionnaireDueDate),
+			documentDue: 'Questionnaire',
+			baseUrl: questionnaireBaseUrl
+		};
+	} else if (isStatementDue(appealCaseData)) {
+		return {
+			deadline: appealCaseData.statementDueDate,
+			dueInDays: calculateDueInDays(appealCaseData.statementDueDate),
+			documentDue: 'Statement',
+			baseUrl: statementBaseUrl
+		};
+	} else if (isFinalCommentDue(appealCaseData)) {
+		return {
+			deadline: appealCaseData.finalCommentsDueDate,
+			dueInDays: calculateDueInDays(appealCaseData.finalCommentsDueDate),
+			documentDue: 'Final comment',
+			baseUrl: finalCommentBaseUrl
+		};
+	} else if (isProofsOfEvidenceDue(appealCaseData)) {
+		return {
+			deadline: appealCaseData.proofsOfEvidenceDueDate,
+			dueInDays: calculateDueInDays(appealCaseData.proofsOfEvidenceDueDate),
+			documentDue: 'Proofs of Evidence',
+			baseUrl: proofsBaseUrl
+		};
+	}
+
+	return {
+		deadline: null,
+		dueInDays: 100000,
+		documentDue: null,
+		baseUrl: null
+	};
+};
+
+// Helper functions, not exported, potential for refactoring as repetitive
```

Comment:
Yep, this feels clunky at the moment. A dictionary of some sort might be a solution? One to consider when looking at the appellant's dashboard and when business logic tied down.

Reply:
Yes something like that - or we stick them in an array and pick the first 'valid' one.

Just thinking - does this take into account the fact that most of these dates can't be overdue? I.e. we shouldn't show statement if the deadline has passed.

Reply:
oh yeah I see.

Could this get "stuck" if they never submit a statement, the due date passes, and it should show the next one?

Reply:
It would no longer get shown in To Do and would move into waiting for review - we were told that only Questionnaires should be shown if overdue.  So if they miss the deadline for any other doc, they can't go further as BO won't accept any other overdue docs

Reply:
Further to discussions, I've amended some logic so that an overdue statement will be skipped and the next document due (if there is one) is then evaluated for display - added another appeal to seed data to cover this scenario